src/algebra/algebra/basic.lean

```diff
@@ -1468,6 +1470,37 @@ by rw [←(one_smul A m), ←smul_assoc, algebra.smul_def, mul_one, one_smul]
 @[simp] lemma algebra_map_smul (r : R) (m : M) : ((algebra_map R A) r) • m = r • m :=
 (algebra_compatible_smul A r m).symm
 
+variables (S A)
+
+@[simp] lemma algebra_map_algebra_map (r : R) : algebra_map S A (algebra_map R S r) =
+  algebra_map R A r :=
+by rw [algebra.algebra_map_eq_smul_one, algebra_map_smul, algebra.algebra_map_eq_smul_one]
+
+@[simp] lemma algebra_map_comp_algebra_map : (algebra_map S A).comp (algebra_map R S) =
+  algebra_map R A :=
+by ext; rw [ring_hom.comp_apply, algebra_map_algebra_map]
+
+variable (R)
+
+/-- `algebra_map S A` as an `R` alg_hom. -/
+def alg_hom_map : S →ₐ[R] A :=
+alg_hom.mk' (algebra_map S A) $ by simp [algebra.algebra_map_eq_smul_one]
```

Comment:
https://github.com/leanprover-community/mathlib/blob/5b75f5ac2a1897f514941f95665f83480959026b/src/algebra/algebra/tower.lean#L112-L114

Also, isn't this just `(algebra.of_id S A).restrict_scalars R`?

Reply:
Yes, but `restrict_scalars` is defined later in the file. I don't think it's a big deal.